src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/ModelAttributes.cs

```diff
@@ -32,17 +33,57 @@ public static IEnumerable<object> GetAttributesForParameter(ParameterInfo parame
         /// <summary>
         /// Gets the attributes for the given <paramref name="property"/>.
         /// </summary>
-        /// <param name="property">A <see cref="ParameterInfo"/> for which attributes need to be resolved.
+        /// <param name="type">The <see cref="Type"/> in which caller found <paramref name="property"/>.
+        /// </param>
+        /// <param name="property">A <see cref="PropertyInfo"/> for which attributes need to be resolved.
         /// </param>
         /// <returns>An <see cref="IEnumerable{object}"/> containing the attributes on the
         /// <paramref name="property"/> before the attributes on the <paramref name="property"/> type.</returns>
-        public static IEnumerable<object> GetAttributesForProperty(PropertyInfo property)
+        public static IEnumerable<object> GetAttributesForProperty(Type type, PropertyInfo property)
```

Comment:
not sure why they aren't already here but please add `[NotNull]` attributes for both parameters to this method and for the parameter to `GetAttributesForType`